### PR TITLE
Bt 720 web sdk readme remove bt client methods document client builder

### PR DESCRIPTION
**Pull request by rick-bt**
BT-720 Update to remove low-value text for BacktraceClient methods, which are well-documented in ts code, and add detail on the ability to create a BacktraceClient with custom file and http handlers.

This initial draft is on /browser. Will update other SDKs when it is ready.

### CODE REVIEW

**Comment by rick-bt** [comment]
@konraddysput - Can you please add code samples for the bottom two features?

File: packages/browser/README.md [hunk 1]
```diff
@@ -340,18 +320,37 @@ The following options are available for the BacktraceClientOptions passed when i
...
+
+// as a BacktraceReport (Error)
+await client.send(new BacktraceReport(new Error('This is a report with a string!')));
+```
+
+### Modify/skip error reports
+A BeforeSend event is triggered when an exception in the managed environment occurs to which you can attach a handler. You can use the BeforeSend event to scrub PII, or extend attributes or JSON object data based on data your application has at the time of exception. A report can be skipped sompletely by returning a null value.
+
+```ts
+  // Code sample
```
Comment (konraddysput) [inline packages/browser/README.md:345]:
```suggestion
const client = BacktraceClient.initialize({
    url: SUBMISSION_URL,
    name: '@backtrace-labs/browser-example',
    version: '0.0.1',
    beforeSend: (data: BacktraceData) => {
        if (data.attributes['do-not-send'] === 'do-not-send') {
            return undefined;
        }
        data.attributes['before-send'] = 'apply-data-in-callback';
        return data;
    },
    userAttributes: {
        'custom-attribute': 'test',
        'custom-annotation': {
            prop1: true,
            prop2: 123,
        },
    },
});

```

File: packages/browser/README.md [hunk 2]
```diff
@@ -340,18 +320,37 @@ The following options are available for the BacktraceClientOptions passed when i
...
-| `initialize(options: BacktraceClientOptions)`                                                                                                   | BacktraceClient | Initializes a new BacktraceClient (returns the same instance on subsequent calls) |
-| `builder(options: BacktraceClientOptions).build()`                                                                                              | BacktraceClient | (Advanced) Sets up a new BacktraceClient for reporting                            |
-| `send(data: BacktraceReport \| Error \| string, reportAttributes: Record<string, unknown> = {}, reportAttachments: BacktraceAttachment[] = [])` | `Promise<void>` | Asynchronously sends error data to Backtrace                                      |
-| `dispose`                                                                                                                                       | void            | Disposes the client                                                               |
+Custom handlers can be implemented to override BacktraceClient file and http operations. Overriding the default operations allows custom encryption for data at rest or in motion.
```
Comment (konraddysput) [inline packages/browser/README.md:350]: In the database we do not have file operations. We do have it in node/react-native/electron/nest.js

File: packages/browser/README.md [hunk 3]
```diff
@@ -340,18 +320,37 @@ The following options are available for the BacktraceClientOptions passed when i
...
 
-### BacktraceReport
+> Do not use these operations to modify the data objects. See [Modify/skip error reports](#modifyskip-error-reports) for the correct method to modify a report before sending it to Backtrace.
 
-A Backtrace Report is the format that ultimately gets sent to Backtrace. Its structure can be found in
-`BacktraceReport.ts`.
+```ts
+  // Code sample
```
Comment (konraddysput) [inline packages/browser/README.md:355]:
```suggestion
const client = BacktraceClient.builder(options)
    .useRequestHandler(requestHandler)
    .useBreadcrumbSubscriber(breadcrumbSubscriber)
    .addAttributeProvider(attributeProvider)
    .build();
```

File: packages/browser/README.md (hunk 2 above)
Comment (konraddysput) [reply packages/browser/README.md:350]: we can leave file operations - maybe someone would like to use a file system in browser by using local storage or something similar. We do not know it yet

**Pull request merged**